src/PrestaShopBundle/Form/Admin/Type/TranslatorAwareType.php

```diff
@@ -38,26 +39,36 @@
     private $translator;
 
     /**
-     * All languages available on shop. Used for translations
+     * Active languages available on shop.
      *
      * @param array $locales
      */
     protected $locales;
 
-    public function __construct(TranslatorInterface $translator, array $locales)
-    {
+    /**
+     * Active and in-active languages available on shop. Used to apply translations
+     *
+     * @var array $allLocales
+     */
+    protected $allLocales;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        ContextLanguageDataProvider $languageDataProvider
+    ) {
         $this->translator = $translator;
-        $this->locales = $locales;
+        $this->locales = $languageDataProvider->getActiveLocales();
+        $this->allLocales = $languageDataProvider->getIncludingInactiveLocales();
```

Comment:
```diff
+ $this->installedLocales = $languageDataProvider->getInstalledLocales();
- $this->locales = $languageDataProvider->getIncludingInactiveLocales();
```

maybe is clearer to use `installedLocales` instead of `allLocales`?